File: GeoMedecins/src/app/_services/medecin.service.ts
import { Injectable } from '@angular/core';
import { Medecin } from '../_models';

@Injectable({
  providedIn: 'root'
})
export class MedecinService {
  medecins = [new Medecin('Gilles', 'Boulon', 'medecin generaliste', 12548, 563214, 'doc1.jpg', 'boulon@mail.com'),
  new Medecin('Celine', 'Truc', 'chirurgienne', 23654, 54896, 'doc2.jpg', 'truc@mail.com')];

  constructor() { }

  getMedecins() {
    return this.medecins;
  }

  getMedecinById(identifiant: number): Medecin {
    console.log(identifiant);
    // for (let i = 0; i < this.medecins.length; i++) {
    //   if (this.medecins[i].id === id) {
    //     return this.medecins[i];
    //   }
    // }
    // return null;
    // Revient à faire :
    return this.medecins.filter(m => m.id === identifiant)[0];
  }
}

File: Spring boot/geomedecins/src/main/java/com/pochard/geomedecins/services/MedecinService.java
package com.pochard.geomedecins.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.pochard.geomedecins.models.Medecin;
import com.pochard.geomedecins.repository.IMedecinRepository;

@Service
public class MedecinService implements IMedecinService {
	@Autowired
	IMedecinRepository iMedecinRepository;

	// private List<Medecin> medecins;

	@Override
	public IMedecinRepository getiMedecinRepository() {
		return iMedecinRepository;
	}

	@Override
	public void setiMedecinRepository(IMedecinRepository iMedecinRepository) {
		this.iMedecinRepository = iMedecinRepository;
	}

	/*
	 * public MedecinService() { /* this.medecins = new ArrayList<>(Arrays.asList(
	 * new Medecin("Gilles", "Boulon", "medecin generaliste", 12548, 563214,
	 * "doc1.jpg", "boulon@mail.com"), new Medecin("Celine", "Truc", "chirurgienne",
	 * 23654, 54896, "doc2.jpg", "truc@mail.com")));
	 */

	public List<Medecin> getAll() {
		return this.iMedecinRepository.findAll();
	}

	@Override
	public Medecin getByID(int identifiant) {
		/*
		 * Medecin med = null; for (Medecin m : medecins) { if (m.getId() ==
		 * identifiant) { med = m; } }
		 */
		return null;
	}

	@Override
	public void addMedecin(Medecin m) {
		// medecins.add(m);
	}

	@Override
	public void editMedecin(Medecin m) {
		/*
		 * for (Medecin med : medecins) { if (m.getId() == med.getId()) {
		 * med.setPrenom(m.getPrenom()); med.setNom(m.getNom());
		 * med.setProfession(m.getProfession()); med.setLattitude(m.getLattitude());
		 * med.setLongitude(m.getLongitude()); med.setPhoto(m.getPhoto());
		 * med.setEmail(m.getEmail()); } }
		 */
	}

	@Override
	public void deleteMedecin(Medecin m) {
		/*
		 * for (Medecin med : medecins) { if (m.getId() == med.getId()) {
		 * medecins.remove(m.getId()); } }
		 */
	}
}
